src/lib.rs

```diff
@@ -1,11 +1,55 @@
+use ipnetwork::{IpNetwork, Ipv4Network, Ipv6Network};
 use rfc7239::{parse, Forwarded, NodeIdentifier, NodeName};
 use std::convert::Infallible;
-use std::iter::once;
+use std::iter::{once, FromIterator, IntoIterator};
 use std::net::{IpAddr, SocketAddr};
 use std::str::FromStr;
 use warp::filters::addr::remote;
 use warp::Filter;
 
+/// Represents a set of IP networks.
+#[derive(Debug, Clone)]
+pub struct IpNetworks {
+    networks: Vec<IpNetwork>,
+}
+
+impl IpNetworks {
+    /// Checks if addr is part of any IP networks included.
+    pub fn contains(&self, addr: &IpAddr) -> bool {
+        self.networks.iter().any(|&network| network.contains(*addr))
+    }
+
+    /// Special constructor that builds IpNetwork from an iterator of IP addresses.
+    pub fn from_ipaddr_iter<'a, T: Iterator<Item = &'a IpAddr>>(addrs: T) -> Self {
```

Comment:
Any reason why this isn't a, `impl FromIterator<&IpAddr>`

Reply:
No, not really.  Let me see.

Reply:
I fixed this a couple days ago.  Can you take a look?